### PR TITLE
fix: fall back to global secrets was not working

### PR DESCRIPTION
**Pull request by matt2e**
fixes https://github.com/TBD54566975/ftl/issues/1339

### CODE REVIEW

File: common/configuration/in_memory_provider.go [hunk 1]
```diff
@@ -18,13 +18,13 @@ func NewInMemoryProvider[R Role]() *InMemoryProvider[R] {
...
 func (p *InMemoryProvider[R]) Role() R     { var r R; return r }
-func (p *InMemoryProvider[R]) Key() string { return "grpc" }
+func (p *InMemoryProvider[R]) Key() string { return "inmemory" }
 
 func (p *InMemoryProvider[R]) Load(ctx context.Context, ref Ref, key *url.URL) ([]byte, error) {
 	if bytes, found := p.values[ref]; found {
 		return bytes, nil
 	}
-	return nil, fmt.Errorf("key %q not found", ref.Name)
+	return nil, ErrNotFound
```
Comment (matt2e) [inline common/configuration/in_memory_provider.go:27]: This specific error must be used

File: common/configuration/manager.go [hunk 2]
```diff
@@ -84,9 +84,8 @@ func (m *Manager[R]) getData(ctx context.Context, ref Ref) ([]byte, error) {
 	key, err := m.resolver.Get(ctx, ref)
 	// Try again at the global scope if the value is not found in module scope.
 	if ref.Module.Ok() && errors.Is(err, ErrNotFound) {
-		gref := ref
-		gref.Module = optional.None[string]()
-		key, err = m.resolver.Get(ctx, gref)
+		ref.Module = optional.None[string]()
```
Comment (matt2e) [inline common/configuration/manager.go:87]: If global config/secret is found, we need the `ref` var to point to it as it is used to access it from the provider.
Previously, resolver would match the global ref, but we would then use the module specific ref with the provider.

File: common/configuration/in_memory_provider.go [hunk 3]
```diff
@@ -18,13 +18,13 @@ func NewInMemoryProvider[R Role]() *InMemoryProvider[R] {
 }
 
 func (p *InMemoryProvider[R]) Role() R     { var r R; return r }
-func (p *InMemoryProvider[R]) Key() string { return "grpc" }
+func (p *InMemoryProvider[R]) Key() string { return "inmemory" }
```
Comment (matt2e) [inline common/configuration/in_memory_provider.go:21]: clean up for an old refactor

**Comment by deniseli** [comment]
Thanks for the comments! Very helpful in understanding the changes :)

**Pull request merged**